swarm/src/dial_opts.rs

```diff
@@ -39,7 +39,8 @@ use std::num::NonZeroU8;
 #[derive(Debug)]
 pub struct DialOpts {
     peer_id: Option<PeerId>,
-    condition: PeerCondition,
+    condition: Option<PeerCondition>,
```

Comment:
Nice, this also works. I was originally thinking of tracking with a `bool` whether this has been set but `Option` is better.